cli/plugins/build/index.js

```diff
@@ -1,47 +1,73 @@
 'use strict';
 
+const fs = require('fs');
+
 /**
  * Build implements Joy commands for static S3 websites, swagger docs and docker images.
- * 
+ *
  */
 class Build {
-
   /**
    * @param {*} joy : Instance of a Joy class c/w config.json, env and helper functions
-   * 
+   *
    */
   constructor(joy) {
     const prog = joy.prog;
     prog
-      .command('build', 'Build a Joy project')
-      .argument('<subcommand>', 'subcommand [staticS3, swagger, docker]', ['staticS3', 'swagger', 'docker'])
+      .command('build_staticS3')
```

Comment:
Currently I am using Caporal for parsing cli commands because it was quick and not really the focus of Joy functionality. But it's not really what we need, we need to write our own parser. The problem is we need a two tier command system. E.g. `build docker`, 'build swagger', 'build site`, etc. Each subcommand has its own set of unique flags, switches and parameters. Caporal doesn't handle this so for short-term only I decided to brute force it with primarycommand_secondarycommand. So, all said and done, I hope we can reach something more like `build static --host s3`